smcontroller/smcontroller.go

```diff
@@ -99,396 +101,241 @@ type CertificateProvider interface {
-	controller.waitAndLock()
-	clients := controller.clients
-	controller.Unlock()
-
-	// we do not support multiple SM right now, set the same config for all SM's
-
-	// Check board config version and set if it is different
-	for _, client := range clients {
-		currentVendorVersion, err := client.getBoardConfigStatus()
-		if err != nil {
-			return aoserrors.Wrap(err)
-		}
-
-		if currentVendorVersion == boardConfig.VendorVersion {
-			continue
-		}
+// SetUnitConfig sets usint config for the node.
+func (controller *Controller) SetUnitConfig(nodeCfg aostypes.NodeConfig, vendorVersion string) error {
+	controller.Lock()
+	defer controller.Unlock()
 
-		if err = client.setBoardConfig(boardConfig); err != nil {
-			return aoserrors.Wrap(err)
-		}
+	handler, err := controller.getNodeHandlerByID(nodeCfg.NodeID)
+	if err != nil {
+		return aoserrors.Wrap(err)
 	}
 
-	return nil
+	return handler.setUnitConfig(nodeCfg, vendorVersion)
 }
 
 // RunInstances runs desired services instances.
-func (controller *Controller) RunInstances(instances []cloudprotocol.InstanceInfo) error {
-	controller.waitAndLock()
-	clients := controller.clients
-	controller.Unlock()
-
-	// we do not support multiple SM right now, ran same instances for all SM's
+func (controller *Controller) RunInstances(nodeID string,
+	services []aostypes.ServiceInfo, layers []aostypes.LayerInfo, instances []aostypes.InstanceInfo,
+) error {
+	controller.Lock()
+	defer controller.Unlock()
 
-	for _, client := range clients {
-		if err := client.runInstances(instances); err != nil {
-			return aoserrors.Wrap(err)
-		}
+	handler, err := controller.getNodeHandlerByID(nodeID)
+	if err != nil {
+		return aoserrors.Wrap(err)
 	}
 
-	return nil
+	return handler.runInstances(services, layers, instances)
 }
 
-// InstallService requests to install servie.
-func (controller *Controller) InstallService(serviceInfo cloudprotocol.ServiceInfo) error {
-	controller.waitAndLock()
-	clients := controller.clients
-	controller.Unlock()
+// InstanceStateAcceptance handles instance state acceptance.
+func (controller *Controller) InstanceStateAcceptance(
+	nodeID string, stateAcceptance cloudprotocol.StateAcceptance,
+) error {
+	controller.Lock()
+	defer controller.Unlock()
 
-	if len(serviceInfo.URLs) == 0 {
-		return aoserrors.New("no service URL")
+	handler, err := controller.getNodeHandlerByID(nodeID)
+	if err != nil {
+		return aoserrors.Wrap(err)
 	}
 
-	var err error
-
-	// we do not support multiple SM right now, install same service for all SM's
-
-	for _, client := range clients {
-		if serviceInfo.URLs[0], err = controller.urlTranslator.TranslateURL(
-			client.cfg.IsLocal, serviceInfo.URLs[0]); err != nil {
-			return aoserrors.Wrap(err)
-		}
-
-		if err = client.installService(serviceInfo); err != nil {
-			return aoserrors.Wrap(err)
-		}
+	if err := handler.instanceStateAcceptance(stateAcceptance); err != nil {
+		return aoserrors.Wrap(err)
```

Comment:
we agreed not to wrap local functions. Change is other places as well.

Reply:
done